### PR TITLE
Restore time remaining option for power applet.

### PR DESCRIPTION
**Pull request by mpetroff**
Cinnamon 2.4 had an option for displaying the estimated battery time remaining in the power applet, but the feature was removed in Cinnamon 2.6. This pull request fixes that regression, fixing #4245.

### CODE REVIEW

**Comment by muzena** [comment]
@mpetroff 
I think this patch is in conflict with this https://github.com/linuxmint/Cinnamon/pull/4427

**Comment by mpetroff** [comment]
This modifies the code immediately before what #4427 modifies.

**Comment by pixunil** [comment]
@muzena, yes they're conflicting (it buts my changes in the middle of @mpetroff's) - but the conflict can only be resolved when one of the PRs is merged.

File: files/usr/share/cinnamon/applets/power@cinnamon.org/applet.js [hunk 1]
```diff
@@ -366,12 +366,29 @@ MyApplet.prototype = {
...
+                            if (this.labelinfo == "nothing") {
+                                ;
                             }
-                            else {
-                                this.set_applet_label("");
+                            else if (this.labelinfo == "time" && seconds != 0) {
+                                let time = Math.round(seconds / 60);
+                                let minutes = Math.floor(time % 60);
+                                let hours = Math.floor(time / 60);
+                                labelText = _("time of battery remaining", "%d:%02d").format(hours,minutes);
```
Comment (pixunil) [inline files/usr/share/cinnamon/applets/power@cinnamon.org/applet.js]: You're wrong with the use of `_(msgid)`, you want to use `C_(msgctx, msgid)` (it point to `Gettext.pgettext`)

Comment (mpetroff) [inline files/usr/share/cinnamon/applets/power@cinnamon.org/applet.js]: Thanks. I originally had `C_` but then saw it only had one other use. I didn't notice the `p` when I glanced at `js/ui/environment.js`, so I thought they were the same function and figured the `C_` form must have been deprecated, so I changed it to `_` (and apparently didn't test it again after making the change).

**Comment by muzena** [comment]
@pixunil I don't understand, I compile and install you repository.
I have wirelles mouse Logitech G700s.
I make patch to upower for G700s/G700: http://cgit.freedesktop.org/upower/commit/?id=305f62adf052aa972523d083ca44d3050f659ec9

Now I look this bug https://github.com/linuxmint/Cinnamon/issues/4329
When I check in applet options Show percentage
In tray icon is only icon without percentage.
Is this work only on laptop battery, or must work on other devices such as a mouse.

**Comment by pixunil** [comment]
@muzena, this isn't a conflict, it is the fault of this [if clause](https://github.com/linuxmint/Cinnamon/pull/4419/files#diff-132ff4b31f6edff30eb6b243f205a330L365), as it only processes Laptop batteries.
Don't know whether this is indented or not.

btw: you can also just swap the files instead of compiling

**Comment by rpbaptist** [comment]
This is actually unrelated to the referenced issue: https://github.com/linuxmint/Cinnamon/issues/4329

**Comment by mpetroff** [comment]
I think I successfully merged in the recent changes to `master`. However, the substantial changes in 2c5b9d5aa3381a1c2e31bdf5f1ee11a7e3574179 break compatibility with the stable 2.6 branch that I'm running, so the post-merge version is untested. As #4427 has been merged, there is no longer a conflict with it.

**Comment by clefebvre** [comment]
Thanks for rebasing this. Successfully tested and about to merge.

**Pull request merged**